carsus/io/cmfgen/util.py

```diff
@@ -0,0 +1,339 @@
+
+
+def find_row(fname, string1, string2=None, how='AND'):
+    """
+    Search for strings in plain text files and returns the matching\
+    line (or row number).
+
+    Parameters
+    ----------
+    fname : str
+        Path to plain text file.
+    string1 : str
+        String to search.
+    string2 : str
+        Secondary string to search (default is None).
+    how : {'OR', 'AND', 'AND NOT'}
+        Search method: `string1` <method> `string2`
+            (default is 'AND').
+
+    Returns
+    -------
+    int, str
+        Returns matching row number and line.
+    """
+
+    if string2 is None:
+        string2 = ''
+
+    with open_cmfgen_file(fname) as f:
+        n = 0
+        for line in f:
+
+            n += 1
+            if how == 'OR':
+                if string1 in line or string2 in line:
+                    break
+
+            if how == 'AND':
+                if string1 in line and string2 in line:
+                    break
+
+            if how == 'AND NOT':
+                if string1 in line and string2 not in line:
+                    break
+
+        else:
+            n, line = None, None
+
+    return n, line
+
+
+def parse_header(fname, keys, start=0, stop=50):
+    """
+    Parse header information from CMFGEN files.
+
+    Parameters
+    ----------
+    fname : str
+        Path to plain text file.
+    keys : list of str
+        Entries to search.
+    start : int
+        First line to search in (default is 0).
+    stop : int
+        Last line to search in (default is 50).
+
+    Returns
+    -------
+    dict
+        Dictionary containing metadata.
+    """
+    meta = {k.strip('!'):None for k in keys}
+
+    with open_cmfgen_file(fname) as f:
+        for line in itertools.islice(f, start, stop):
+            for k in keys:
+                if k.lower() in line.lower():
+                    meta[k.strip('!')] = line.split()[0]
+
+    return meta
+
+
+def get_seaton_phixs_table(threshold_energy_ryd, sigma_t, beta, s, nu_0=None, n_points=1000):
+    """
+    References:
+
+        Atomic data for opacity calculations. I. General description
+
+        Seaton, M. J.
+
+        Journal of Physics B: Atomic, Molecular, and Optical Physics,
+        Volume 20, Issue 23, pp. 6363-6378 (1987).
+    """
+    energy_grid = np.linspace(0.0, 1.0, n_points, endpoint=False)
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    for i, c in enumerate(energy_grid):
+        energy_div_threshold = 1 + 20 * (c ** 2)
+
+        if nu_0 is None:
+            threshold_div_energy = energy_div_threshold ** -1
+            cross_section = sigma_t * (beta + (1 - beta) * threshold_div_energy) * (threshold_div_energy ** s)
+
+        else:
+            threshold_energy_ev = threshold_energy_ryd * RYD_TO_EV
+            energy_offset_div_threshold = energy_div_threshold + (nu_0 * 1e15 * H_IN_EV_SECONDS) / threshold_energy_ev
+            threshold_div_energy_offset = energy_offset_div_threshold ** -1
```

Comment:
```suggestion
            threshold_div_energy_offset = energy_div_threshold**-1 * (1 + (nu_0 * 1e15 * H_IN_EV_SECONDS) / threshold_energy_ev)
```
I think the offset should be added to the ionization edge instead of the current energy. See the [CMFGEN source code](https://github.com/juliotux/cmfgen/blob/6cae33c174a561fe26ec9fb13efd55345b5fe4e1/newsubs/sub_phot_gen.f#L482).

Reply:
Thanks for finding this. I suggest renaming threshold_div_energy_offset to offset_threshold_div_energy just to be slightly more clear.

Reply:
That's a good point. I have changed it.